### ISSUE TITLE
docker:  conflict between conda and pip

### ISSUE THREAD

**Issue by KrisThielemans** [SyneRBI/SIRF-SuperBuild#655]
We ~k~now install CIL via conda
https://github.com/SyneRBI/SIRF-SuperBuild/blob/a3bc245befefd35c310c7312c8263bb0949afab6/docker/user_python-ubuntu.sh#L42
However, when building a new Docker image, I get
```
/conda install --yes -c conda-forge -c intel -c ccpi -c astra-toolbox/label/dev --file requirements_conda_forge.txt
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##
  environment location: /opt/pyvenv
  added / updated specs:
    - ccpi-regulariser
    - cil
    - cil-astra
    - tigre
    - tomophantom==1.4.10
The following packages will be downloaded:
<snip>
The following packages will be DOWNGRADED:
  openssl                                  3.0.0-h7f98852_2 --> 1.1.1l-h7f98852_0
  python                          3.9.10-hc74c709_2_cpython --> 3.9.10-h85951f9_2_cpython
```
The message about downgrading Python seems problematic to me, especially as we have done some [`pip install` stuff earlier](https://github.com/SyneRBI/SIRF-SuperBuild/blob/a3bc245befefd35c310c7312c8263bb0949afab6/docker/user_python-ubuntu.sh#L37)

conda recommends creating an environment with all the packages listed to avoid such version conflicts (but I don't think we're using a virtual env in the conda case). I think at the very least, the `pip install` lines should be moved after the `conda install`.

We also do
https://github.com/SyneRBI/SIRF-SuperBuild/blob/a3bc245befefd35c310c7312c8263bb0949afab6/docker/user_python-ubuntu.sh#L44
which I wonder if is a great idea with dependencies etc

**Comment by KrisThielemans** [comment]
By the way, to me it looks like `build_python-ubuntu.sh` is obsolete when using `conda`.

**Comment by KrisThielemans** [comment]
I've created a custom docker to be able to play around with this, skipping the `user_python-ubuntu.sh` (and SIRF installation itself). Sadly, executing the first and any of the next 2 in any order
```
$ conda install --yes -c conda-forge -c intel -c ccpi -c astra-toolbox/label/dev -n --file requirements_conda_forge.txt 
$ conda update -c conda-forge -y setuptools pip
$ pip install -U -r requirements.txt
```
results in
```
$ python
bash: /opt/pyvenv/bin/python: Invalid argument
```
(python links to python3.9, but that gives the same error message)

**Comment by KrisThielemans** [comment]
By the way, it's possible to put `pip` packages in ` conda` `environment.yml`. See https://python.engineering/35245401-combining-conda-environment-yml-with-pip-requirements-txt/. This might be the best way. I haven't tried this.

**Comment by paskino** [comment]
I think in the first command `-n` requires a parameter, i.e. the name of the environment you want to install the packages into. In your command this is absent.

**Comment by KrisThielemans** [comment]
oops. sorry. I tried a few things, including using `conda create -n myenv ...` but it didn't seem to make any difference, so I reported only `conda install` (but forgot to remove the `-n`)

**Comment by KrisThielemans** [comment]
Probably we're missing a `conda activate` before the `pip install`?

**Comment by paskino** [comment]
Is this issue still relevant?

**Comment by KrisThielemans** [comment]
> Is this issue still relevant?

no idea. I have no time to check. If you don't experience any problems with building docker/VM etc, then I would postpone it to the next milestone.

**Comment by KrisThielemans** [comment]
at least try to move https://github.com/SyneRBI/SIRF-SuperBuild/blob/a3bc245befefd35c310c7312c8263bb0949afab6/docker/user_python-ubuntu.sh#L35-L38 to the end of the file

**Comment by paskino** [comment]
I believe this may be closed by #787 ?
